src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5373,13 +5374,28 @@ int wm_vuldet_generate_os_and_kernel_package(sqlite3 *db, agent_software *agent)
                 product_name[2] = "macos";
             }
         break;
+        case FEED_AL1:
+            product_name[0] = "linux_1";
+            product_name[1] = V_KERNEL;
+            vendor = V_AMAZON;
+        break;
+        case FEED_AL2:
+            product_name[0] = "linux_2";
+            vendor = V_AMAZON;
+        break;
         default:
             return 1;
     }
 
     for (int i = 0; product_name[i] != NULL; i++) {
         // Generate os package
-        if (agent->os_release) {
+        if (agent->dist == FEED_ALAS){
```

Comment:
We could replace the value of `agent->os_release` with `*`. That way we avoid repeating this piece of code.